lib/rspec/mocks/exclude_stubbed_classes_from_subclasses.rb

```diff
@@ -0,0 +1,56 @@
+module RSpec
+  module Mocks
+    # Support for `exclude_stubbed_classes_from_subclasses` configuration.
+    #
+    # @private
+    class ExcludeStubbedClassesFromSubclasses
+      class << self
+        def enable!
+          return unless RUBY_VERSION >= "3.1"
+          return if Class.respond_to?(:subclasses_with_rspec_mocks)
+
+          Class.class_eval do
+            def subclasses_with_rspec_mocks
+              subclasses_without_rspec_mocks - RSpec::Mocks::ExcludeStubbedClassesFromSubclasses.excluded_subclasses
+            end
+
+            alias subclasses_without_rspec_mocks subclasses
+            alias subclasses subclasses_with_rspec_mocks
+          end
+        end
+
+        def disable!
+          @excluded_subclasses = []
+
+          if Class.respond_to?(:subclasses_with_rspec_mocks)
+            Class.class_eval do
+              undef subclasses_with_rspec_mocks
+              alias subclasses subclasses_without_rspec_mocks
+              undef subclasses_without_rspec_mocks
+            end
+          end
+        end
+
+        def excluded_subclasses
+          require 'weakref' unless defined?(::WeakRef)
+
+          @excluded_subclasses ||= []
+          @excluded_subclasses.select(&:weakref_alive?).map do |ref|
+            begin # rubocop:disable Style/RedundantBegin
```

Comment:
It slipped my mind, isn’t it possible to squash that so that rescue is on the map itself?

Reply:
Ah, nevermind, Ruby 1.8 probably doesn’t support that. Do you mind making a note to indicate that?

Reply:
I removed the comment because the CI was failing because of that (I don't have the same behavior locally with the same version of Ruby/Rubocop).